subsys/partition_manager/CMakeLists.txt

```diff
@@ -12,9 +12,13 @@ set_property(GLOBAL APPEND PROPERTY
   -DUSE_PARTITION_MANAGER=$<TARGET_EXISTS:partition_manager>
   )
 
+# Safeguard to make sure folder naming does not break building on Windows
+if(${PROJECT_BINARY_DIR} MATCHES "[<>|?*]")
+  message(FATAL_ERROR "Characters <>|?* are not allowed as file or folder name")
```

Comment:
Let's de-reference PROJECT_BINARY_DIR to show the user which file had the offending character.

Reply:
Good idea it'll give a better indication to the user where to look :)

Reply:
Now the error message will look like this:

```
CMake Error at /Users/sigvartmh/Programming/multi/nrf/subsys/partition_manager/CMakeLists.txt:17 (message):
  The character "|" is not allowed as filename or folder name.

  The offending path is:
  $/Users/sigvartmh/Programming/multi/zephyr/samples/hello_world/error|/spm/zephyr
```

Which I think looks pretty nice.